lib/practice_exercises.rb

```diff
@@ -1,13 +1,78 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) n being all elements in the list. They are get iterated through
+#compared. 
+# Space Complexity: ?Would be O(1) the space stays the same throughout. No new arrays 
+# are created
+
+#compare the first number to all numbers until you reach a num thats not equal
+#Then replace taht first num with num that is different, and add to the counter
+#Once you reach the end of the array return the counter.
+
+
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  if list.length == nil
+    return []
+  elsif list[0] == []
+    return []
+  end
+  
+  compare_to = list[0]
+  i = 0
+  counter = 0
+  
+  while list.length >= i
+    if compare_to == list[i]
+      list.delete_at(list[i])
+    elsif compare_to != list[i]
+      counter += 1
+      compare_to = list[i]
+    end
+    i += 1
+  end
+  return list
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+
+# Time Complexity: O(n^2)Each word gets iterated through and every letter in the word
+# gets iterated through. 
+# Space Complexity: No new Array is crated except common word. 
 def longest_prefix(strings)
-  raise NotImplementedError, "Not implemented yet"
+
+  common_word = ""
```

Comment:
You are going to need 2 loops here one of which is looping through the characters inside one of the strings.  You only have one loop here and you're comparing using an index for the array as an index for a letter in a word.